src/main/java/org/hyperskill/hstest/checker/CheckLibraryVersion.java

```diff
@@ -0,0 +1,129 @@
+ */
+public class CheckLibraryVersion {
+
+    private String VERSION_FILE = "src/main/java/org/hyperskill/hstest/resources/version.txt";
+
+    private String currentVersion;
+    private String latestVersion;
+    private boolean isLatestVersion = true;
+
+    public CheckLibraryVersion() {
+    }
+
+    /**
+     * Checks if the current version of the library is the latest one on GitHub releases page of the library.
+     * If not, throws an exception.
+     */
+    public void checkVersion() {
+        LocalDate lastChecked = null;
+        File lastCheckedFile = new File("LastChecked.txt");
+        if (lastCheckedFile.exists()) {
+            try {
+                FileReader fileReader = new FileReader(lastCheckedFile);
+                BufferedReader reader = new BufferedReader(fileReader);
+                lastChecked = LocalDate.parse(reader.readLine());
+                reader.close();
+            } catch (IOException e) {
+                throw new RuntimeException(e);
+            }
+        }
+
+        if (lastChecked != null && lastChecked.equals(LocalDate.now())) {
+            return;
+        }
+
+        InputStream inputStream = getClass().getClassLoader().getResourceAsStream(VERSION_FILE);
+        if (inputStream == null) {
+            throw new RuntimeException("Unable to find hs-test library version file");
+        }
+
+        try {
+            currentVersion = new BufferedReader(new InputStreamReader(inputStream)).readLine();
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+        latestVersion = getLatestHsTestVersionFromGitHub();
+
+        if (!currentVersion.equals(latestVersion)) {
+            if (currentVersion.split("\\.")[0] != latestVersion.split("\\.")[0]) {
+                throw new RuntimeException(getFeedback());
+            } else {
+                isLatestVersion = false;
+            }
+        }
+
+        lastChecked = LocalDate.now();
+        try (FileWriter writer = new FileWriter(lastCheckedFile)) {
+            writer.write(lastChecked.toString());
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    /**
+     * Returns latest version of the library from GitHub releases page of the library.
+     * @return String latest version of the library
+     */
+    private String getLatestHsTestVersionFromGitHub() {
+        HttpURLConnection connection = null;
+        try {
+            URL url = new URL("https://api.github.com/repos/hyperskill/hs-test/releases/latest");
+            connection = (HttpURLConnection) url.openConnection();
+            connection.setRequestMethod("GET");
+            connection.setRequestProperty("Accept", "application/vnd.github+json");
+
+            int responseCode = connection.getResponseCode();
+            if (responseCode != 200) {
+                throw new UnexpectedError("Error getting latest version of hs-test library from GitHub");
```

Comment:
Ok, in case GitHub is down our users can't even test their code. It is not OK. I think, in case we can't check library version, you should assume it's the latest version.

Reply:
Also, keep in mind that online checking (from the website) is done in the isolated environment and there are no internet connection. Such way of checking should continue working fine. We should not rely on internet being accessible whole checking user code. Could you please tell what is a motivation behind this issue?

Reply:
I think you mean that with online verification it will not be possible to get the current version in GitHub, the latestVersion will be equal to the currentVersion and the user will continue to work with tests.
The relevance of the library to work in images for online verification is not the task of the user, we must maintain the relevance of the library for online verification.

Reply:
@polischuks Yes. But because online checking in done without access to the internet (Docker image has no access to he internet) and that is why the check will fail. That's what I mean